src/centralized_telescope/proof.rs

```diff
@@ -36,10 +36,66 @@ impl Proof {
     ///
     /// A `Proof` structure
     pub(super) fn new(set_size: u64, params: &Params, prover_set: &[Element]) -> Option<Self> {
+        debug_assert!(crate::utils::misc::check_distinct(prover_set));
+
+        Self::prove_routine(set_size, params, prover_set).1
+    }
+
+    /// Alba's proving algorithm used for benchmarking, returning a proof if found
+    /// as well as the number of steps done when generating it.
+    ///
+    /// # Arguments
+    ///
+    /// * `set_size` - the size of the prover set to lower bound
+    /// * `params` - the internal parameters to generate a proof from
+    /// * `prover_set` - the dataset to generate a proof from
+    ///
+    /// # Returns
+    ///
+    /// The number of steps, and `Some(Proof)` structure
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use alba::centralized_telescope::params::Params;
+    /// use alba::centralized_telescope::proof::Proof;
+    /// let set_size = 200;
+    /// let params = Params::new(128.0, 128.0, 1_000, 750);
+    /// let mut prover_set = Vec::new();
+    /// for i in 0..set_size {
+    ///     prover_set.push([(i % 256) as u8 ; 48]);
+    /// }
+    /// let (setps, proof_opt) = Proof::bench(set_size, &params, &prover_set);
+    /// ```
+    pub fn bench(set_size: u64, params: &Params, prover_set: &[Element]) -> (u64, Option<Proof>) {
```

Comment:
Something happened to the `internal` module. I think the long documentation above is not necessary since this function is not supposed to be used.

Reply:
We cannot have internal modules in a impl unfortunately. As the function needs to be accessed, it is public hence I added the doc for public functions.

Reply:
I see. Then suggest replace the long function doc with "Alba's proving algorithm used for benchmarking, returning a proof if found as well as the number of steps done when generating it. Only for internal usage. Do not use." since it's not meant to be used publicly.